docs/Migrating.md

```diff
@@ -11,23 +11,28 @@ The versions can be found in
 * Curriculum config files are now YAML formatted and all curricula for a training run are combined into a single file.
-* The `--num-runs` command-line option has been removed.
-* The "Reset on Done" setting in AgentParameters was removed; this is now effectively always true. `AgentOnDone` virtual method on the Agent has been removed.
-* Agents will always request a decision after being marked as `Done()` and will no longer wait for the next call to `RequestDecision()`.
-* The `agentParameters` field of the Agent has been removed. (Contained only `maxStep` information)
-* `maxStep` is now a public field on the Agent. (Was moved from `agentParameters`)
-* The `Info` field of the Agent has been made private. (Was only used internally and not meant to be modified outside of the Agent)
-* The `GetReward()` method on the Agent has been removed. (It was being confused with `GetCumulativeReward()`)
-* The `AgentAction` struct no longer contains a `value` field. (Value estimates were not set during inference)
-* The `GetValueEstimate()` method on the Agent has been removed.
-* The `UpdateValueAction()` method on the Agent has been removed.
-* Calling `Done()` on the Agent will now reset it immediately and call the `AgentReset` virtual method. (This is to simplify the previous logic in which the Agent had to wait for the next `EnvironmentStep` to reset)
+* The `--num-runs` command-line option has been removed from `mlagents-learn`.
+* Several fields on the Agent were removed or made private in order to simplify the interface.
+  * The `agentParameters` field of the Agent has been removed. (Contained only `maxStep` information)
+  * `maxStep` is now a public field on the Agent. (Was moved from `agentParameters`)
+  * The `Info` field of the Agent has been made private. (Was only used internally and not meant to be modified outside of the Agent)
+  * The `GetReward()` method on the Agent has been removed. (It was being confused with `GetCumulativeReward()`)
+  * The `AgentAction` struct no longer contains a `value` field. (Value estimates were not set during inference)
+  * The `GetValueEstimate()` method on the Agent has been removed.
+  * The `UpdateValueAction()` method on the Agent has been removed.
+* The deprecated `RayPerception3D` and `RayPerception3D` classes were removed, and the `legacyHitFractionBehavior` argument was removed from `RayPerceptionSensor.PerceiveStatic()`.
 
 ### Steps to Migrate
+* If your Agent implemented `AgentOnDone` and did not have the checkbox `Reset On Done` checked in the inspector, you must call the code that was in `AgentOnDone` manually.
+* If you give your Agent a reward or penalty at the end of an episode (e.g. for reaching a goal or falling off of a platform), make sure you call `AddReward()` or `SetReward()` *before* calling `Done()`. Previous
```

Comment:
~. Previous~

Reply:
oops, that meant to write "Previously, the order didn't matter."

Reply:
Updated